go-controller/pkg/ovn/egressgw.go

```diff
@@ -631,7 +632,7 @@ func getExternalIPsGR(watchFactory *factory.WatchFactory, nodeName string) ([]*n
 // deletePodSNATOps creates ovsdb operation that removes per pod SNAT rules towards the nodeIP that are applied to the GR where the pod resides
 // used when disableSNATMultipleGWs=true
 func deletePodSNATOps(nbClient libovsdbclient.Client, ops []ovsdb.Operation, gwRouterName string, extIPs, podIPNets []*net.IPNet) ([]ovsdb.Operation, error) {
-	nats, err := buildPodSNAT(extIPs, podIPNets)
+	nats, err := buildPodSNAT(extIPs, podIPNets, "")
```

Comment:
deletePodSNATOps is used by secondary network controller. Why do you pass ""? Looking closer at this,

func isEquivalentNAT(existing *nbdb.NAT, searched *nbdb.NAT) bool {

looks like it is not comparing match criteria. I wonder if that was something missed by the conditional snat update in ovnk?

Reply:
You are right, I modified `isEquivalentNAT` to take the match into account and fixed the delete path.